yogstation/code/modules/antagonists/shadowling/shadowling_abilities.dm

```diff
@@ -0,0 +1,878 @@
+	human_req = TRUE
+	clothes_req = FALSE
+	action_icon_state = "glare"
+	action_icon = 'yogstation/icons/mob/actions.dmi'
+
+/obj/effect/proc_holder/spell/targeted/sling/glare/InterceptClickOn(mob/living/caller, params, atom/t)
+	. = ..()
+	if(!target)
+		return
+	if(target.stat)
+		to_chat(usr, "<span class='warning'>[target] must be conscious!</span>")
+		revert_cast()
+		return
+	if(is_shadow_or_thrall(target))
+		to_chat(usr, "<span class='warning'>You cannot glare at allies!</span>")
+		revert_cast()
+		return
+	var/mob/living/carbon/human/M = target
+	usr.visible_message("<span class='warning'><b>[usr]'s eyes flash a purpleish-red!</b></span>")
+	var/distance = get_dist(target, usr)
+	if (distance <= 4)
+		target.visible_message("<span class='danger'>[target] suddendly collapses...</span>")
+		to_chat(target, "<span class='userdanger'>A purple light flashes across your vision, and you lose control of your movements!</span>")
+		target.Knockdown(100)
+		M.silent += 10
+	else //Distant glare
+		var/loss = 120 - (distance * 10)
+		target.adjustStaminaLoss(loss)
+		target.stuttering = loss
+		to_chat(target, "<span class='userdanger'>A purple light flashes across your vision, and exhaustion floods your body...</span>")
+		target.visible_message("<span class='danger'>[target] looks very tired...</span>")
+	charge_counter = 0
+	start_recharge()
+	remove_ranged_ability()
+	usr = null
+	target = null
+
+/obj/effect/proc_holder/spell/aoe_turf/veil //Puts out most nearby lights except for flares and yellow slime cores
+	name = "Veil"
+	desc = "Extinguishes most nearby light sources."
+	panel = "Shadowling Abilities"
+	charge_max = 150 //Short cooldown because people can just turn the lights back on
+	human_req = TRUE
+	clothes_req = FALSE
+	range = 5
+	action_icon_state = "veil"
+	action_icon = 'yogstation/icons/mob/actions.dmi'
+	var/admin_override = FALSE //Requested by Shadowlight213. Allows anyone to cast the spell, not just shadowlings.
+
+/obj/effect/proc_holder/spell/aoe_turf/proc/extinguishItem(obj/item/I, cold = FALSE) //Does not darken items held by mobs due to mobs having separate luminosity, use extinguishMob() or write your own proc.
+	var/blacklisted_lights = list(/obj/item/flashlight/flare, /obj/item/flashlight/slime, /obj/item/electronic_assembly)
+	if(istype(I, /obj/item/flashlight))
+		var/obj/item/flashlight/F = I
+		if(F.on)
+			if(cold)
+				if(is_type_in_list(F, blacklisted_lights))
+					F.visible_message("<span class='warning'>The sheer cold shatters [F]!</span>")
+					qdel(F)
+				else
+					return
+			if(is_type_in_list(I, blacklisted_lights))
+				I.visible_message("<span class='danger'>[I] dims slightly before scattering the shadows around it.</span>")
+				return F.brightness_on //Necessary because flashlights become 0-luminosity when held.  I don't make the rules of lightcode.
+			F.on = FALSE
+			F.update_brightness()
+	else if(istype(I, /obj/item/pda))
+		var/obj/item/pda/P = I
+		P.fon = FALSE
+	I.set_light(0)
```

Comment:
Same thing I told guppy, blacklisting the assembly is useless because it never passes the check. Moreover, I.set_light(0) doesn't work for assemblies because you can have a ticker constantly reenable it.

Reply:
It does disable assembly lights. And if someone has enough brain cells to make a light that turns itself back on, they deserve to have an edge.

Reply:
I'm not saying it doesn't disable them. I'm saying it doesn't work because assembly is a hard counter to slings. All it takes is a battery, a light, and a ticker. Connect the output of the ticker to the light pulse, enable, and voila, goodbye shitlings. Which means the current sling meta would be to kill circuits first and science.

Reply:
Also, the blacklisted assembly does nothing at all, so might as well remove it.